snowpack/src/scan-imports.ts

```diff
@@ -156,10 +157,21 @@ function parseFileForInstallTargets({
   contents,
 }: SnowpackSourceFile): InstallTarget[] {
   try {
-    if (baseExt === '.css' || baseExt === '.scss' || baseExt === '.sass' || baseExt === '.less') {
-      return parseCssForInstallTargets(contents);
-    } else {
-      return parseJsForInstallTargets(contents);
+    switch (baseExt) {
+      case '.css':
+      case '.less':
+      case '.sass':
+      case '.scss': {
+        return parseCssForInstallTargets(contents);
+      }
+      case '.html':
+      case '.svelte':
+      case '.vue': {
+        return parseJsForInstallTargets(extractJSFromHTML({contents, baseExt}));
```

Comment:
So this is a confusing change to follow (I’m a little confused myself), but a short version is: we have 2 functions in this file that are semi-responsible for scanning for import statements:

- `scanImports()`
- `parseFileForImportTargets()`

For the case of HTML, it was actually broken here but we didn’t notice because we were parsing the whole thing as JS and catching things inside `<script>` tags.

It’s hard to explain this change in further detail, but basically the point at which we stripped out the `<script type="module">` tags from the HTML was hitting some files and missing others, so some files were being properly transformed while some weren’t (could be because both `scanImports` and `parseFileForInstallTargets` are both exported, and one calls the other, so it became possible to bypass a code path in certain conditions).

This PR changes behavior so now each HTML’s `<script type="module">` tags are extracted properly, and at the right point.

Ditto for `.svelte` and `.vue` files.